discovery-provider/src/api/v1/tracks.py

```diff
@@ -413,8 +433,29 @@ def get(self, track_id):
         if track["is_delete"] or track["user"]["is_deactivated"]:
             abort_not_found(track_id, ns)
 
+        request_args = stream_parser.parse_args()
+
+        # signature for the track to be included as a query param in the redirect to CN
+        signature = get_track_stream_signature(
+            {
+                "track": track,
+                "user_data": request_args.get("user_data", None),
```

Comment:
nit: Seems like the user should be signing the rest of the request body data i.e. track and prem sig (if present) w/ timestamp, not some arbitrary data